### PR TITLE
Add new methods

### PR DESCRIPTION
**Pull request by Flocqst**
## Description
For each 128 type add `divToDimensionless`, `ceilDivToDimensionless`, `greaterThanZero`, `lessThanZero`, `greaterThanOrEqualToZero` &`lessThanOrEqualToZero`

For each 256 type  add `divToDimensionless`, `ceilDivToDimensionless` & `greaterThanZero`

Add tests for each of these methods.

## Related issue(s)
//

## Motivation and Context
Methods currently not covered by quanto-dimensions

### CODE REVIEW

File: src/Int128/BaseInt128/ValueType.sol [hunk 1]
```diff
@@ -148,6 +148,45 @@ function isZero(BaseInt128 x) pure returns (bool) {
...
+{
+    result = x.unwrap() / y.unwrap();
+}
+
+/// @notice Implements the ceiling division of BaseInt128 by BaseInt128 to dimensionless.
+function ceilDivide(BaseInt128 x, BaseInt128 y) pure returns (int128) {
+    return x.unwrap() / y.unwrap()
+        + (
+            (
+                ((x.unwrap() < 0) != (y.unwrap() < 0))
```
Comment (tommyrharper) [inline src/Int128/BaseInt128/ValueType.sol:164]: **Question:** I am confused as to the reasoning behind this logic? Could you explain.

Comment (Flocqst) [reply src/Int128/BaseInt128/ValueType.sol:164]: Here's the reasoning :

For unsigned int, it is straightforward :
 ```x / y + (x % y == 0 ?  0 : 1)```
You make the div and then you add 1 if there is a remainder to make the ceiling operation.

For signed int, you have the same logic for positive numbers but it is different for negative numbers:
```x / y + ( (    ((x < 0) != (y < 0))    ||    (x % y) == 0 ) ? 0 : 1);```

Indeed, if you make the div of a negative number by a positive number (or vice versa), you already have the ceiling of the div ( e.g. `-10/3 = -3.3333` so the ceiling is `-3`,  which is what you get when you make the division with int types)

Which is why you add to the first formula the following check :
```((x < 0) != (y < 0))```
Which checks if the result of the division is a negative number, in which case you add nothing (even if there is a remainder).

Comment (tommyrharper) [reply src/Int128/BaseInt128/ValueType.sol:164]: I think I found an edge case where this fails:
```
        x = BaseInt128.wrap(-10);
        y = BaseInt128.wrap(-3);
        result = x.ceilDivide(y);
        assertEq(result, 3); // fails here as result is 4
```

Comment (tommyrharper) [reply src/Int128/BaseInt128/ValueType.sol:164]: Got it! Makes sense, great work 💪

**Review by tommyrharper — APPROVED** [review]
Awesome work again! Nice one @Flocqst 🔥 🏅

**Pull request merged**